gmaps/maps.py

```diff
@@ -30,7 +30,108 @@ def _config_default(self):
         return _default_configuration
 
 
+# I assume (lng0, lng1) in a layer's
+# bounds means that the layer wants the map to
+# cover all the longitudes that lie
+# (going eastwards) from lng0 to lng1.
+# For example: (170,-170) means we cover 20 degrees of
+# latitude starting at 170 (through 180)
+# Whereas (-170,170) means we start where the previous
+# interval ended (at 170), go through lng 0, and eventually
+# to 170 (340 degrees).
+#
+# As a side note, this definition means that (lng1,lng0) is
+# the complement of (lng0,lng1) on this notation.
+# It is important we have this direction information somehow,
+# because otherwise the bounds are underspecified.
+#
+# As another side note, the meaning of (15, 30) in this notation
+# is the same as that of (15+/-360, 30) and that of (15, 30 +/- 360),
+# because we stop as soon as we hit the coordinate the first
+# time.
+#
+# Note: the google api for setting map bounds seems to follow this
+# convention as well, but I'm not sure gmaps overall does.
+# (At least, I couldn't verify)
+#
+# -180                                            180 = -180
+#   |     >--------------->                        |
+#   |         >------->                            |
+#   |            >------------->                   |
+#   |->    >------->                         >-----|
+#   11000012223334433322221111100000000000000111111|
+#                              ^ biggest empty range
+# the bounds are then
+#  First we find which longitude ranges are feasible
+#  to cut (they have no range overlaps) by counting number of
+#  intervals that overlap and finding 0s.
+#  The initial condition (at -180)
+#  we find from counting how many intervals have west > east.
+#  then we go through our list.
+#  Of all the segments with a 0 count, the longest one
+#  is the one we should use to cut the map (here we
+#  have two choices)
+
+#
+def normalize_lng(lng):
+    """ returns an equivalent longitude in the [-180,180) range """
+    lng = lng % 360
+    if lng >= 180:
+        lng = lng - 360
+
+    assert(lng >= -180)
+    assert(lng < 180)
+    return lng
+
+
+# Then we can search for the best gap, including ones that wrap around
+# we achieve this by sorting and counting covering intervals
+# 0 means no bounds overalp with that segment
+def get_lng_bound(bounds_list):
+
+    # extract and normalize lngs from bounds (if they arent normalized)
+    directed_intervals = [(normalize_lng(b[0][1]),
+                           normalize_lng(b[1][1]))
+                          for b in bounds_list]
+
+    # coverage is initially the number of wrap-around intervals. it can be 0.
+    coverage = sum([1 for (lng0, lng1)
+                    in directed_intervals if lng0 > lng1])
```

Comment:
So about this last comment,  The google bounds API already interprets (x,x) to mean just that point. To express the concept of the full circumference one has the option of using (x, x - epsilon). I think there is value for users and developers in trying to keep the python semantics consistent with their API.

Even if we go against the google grain,  there would need to be a way to express the 1 point bounds concept.  If we take (x,x) to mean wrapping around the entire world, then we need to come up something for the other concept.